CMakeLists.txt

```diff
@@ -67,11 +67,6 @@ if (BUILD_DEPS)
         if (CRT_FIPS)
             list(APPEND AWSLC_CMAKE_ARGUMENTS -DFIPS=ON)
             list(APPEND AWSLC_CMAKE_ARGUMENTS -DDISABLE_PERL=OFF)
-
-            if (DEFINED ENV{GO_PATH})
-                list(APPEND AWSLC_CMAKE_ARGUMENTS -DGO_EXECUTABLE=$ENV{GO_PATH}/go)
-                message(STATUS "Overriding GO_EXECUTABLE to ${GO_EXECUTABLE}")
-            endif()
```

Comment:
This wasn't causing problems, but I don't know why it's here, so removing it.

[GOPATH](https://go.dev/wiki/GOPATH) is a thing in Golang, but this one has an underscore in it??? GOPATH would point at the directory containing `bin/go`, but this one would point at the directory containing `go/bin/go`???

I believe this was some way of communicating with builder.json, that is no longer in use. Here are some commits from the PR that added these lines:
- GO_PATH added to builder.json and CMakeLists.txt here: https://github.com/awslabs/aws-crt-java/pull/772/commits/ed809e43c20f7c72e0ff2c8e93db2f551427dcaf
- GO_PATH tweaked in builder.json here https://github.com/awslabs/aws-crt-java/pull/772/commits/c11cf464494ee7e64dd90711cb7ceb152e4e42a5
- GO_PATH removed from builder.json here: https://github.com/awslabs/aws-crt-java/pull/772/commits/96c5e1e059d467af743e4bd8a7d1bab9bbc1bd11

Reply:
https://github.com/awslabs/aws-crt-builder/pull/267/files#diff-f0fd7d35e67059aa0e19b135ded7e70e89b146b793db24528f9f8bda7c07b855R80-R84

Reply:
This is used for cross compile, we need to install the go executable for the target platform and override the executable path for the aws-lc to find the right go executable to use. 

so, we still need this to override the go_path if it's cross compile. We could add more comments to make it more clear.

Reply:
I put this back, and added comment explaining it